server/config/config.go

```diff
@@ -196,6 +199,28 @@ type StateSyncConfig struct {
 	SnapshotKeepRecent uint32 `mapstructure:"snapshot-keep-recent"`
 }
 
+type (
+	// StoreConfig defines application configuration for state streaming and other
+	// storage related operations.
+	StoreConfig struct {
```

Comment:
still a bit confused how this gets fed into the the streamer? is it viper magic?

Reply:
streamer just do `appOpts.Get("store.streamers")`

Reply:
Yeah the magic is already there (granted that too could use some cleanup), but this PR just sets up the default config and template so it's clear to devs how to structure the config.